uk.co.compucorp.civicrm.civihrsampledata/tests/phpunit/CRM/CiviHRSampleData/Importer/AbsencePeriodTest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+require_once EXTENSION_ROOT_DIR . 'CRM/CiviHRSampleData/Importer/AbsencePeriod.php';
+
+/**
+ * Class CRM_CiviHRSampleData_Importer_AbsencePeriodTest
+ *
+ * @group headless
+ */
+class CRM_CiviHRSampleData_Importer_AbsencePeriodTest extends CRM_CiviHRSampleData_BaseImporterTest {
+
+  private $rows;
+
+  public function setUpHeadless() {
+    return \Civi\Test::headless()
+      ->install('org.civicrm.hrabsence')
+      ->apply();
+  }
+
+  public function setUp() {
+    $this->rows = [];
+    $this->rows[] = $this->importHeadersFixture();
+  }
+
+  public function testImport() {
+    $this->rows[] = [
+      '2016',
+      '2016 (Jan 1 to Dec 31)',
+      '2016-01-01 02:00:00',
+      '2017-01-01 01:59:59',
+    ];
+
+    $this->runImporter('CRM_CiviHRSampleData_Importer_AbsencePeriod', $this->rows);
+
+    $this->assertEquals('2016', $this->apiQuickGet('HRAbsencePeriod','name', '2016'));
```

Comment:
but I agree to (what about negative tests?) & ( What if the row doesn't have the right number of columns? )  ... my comment above apply to your questions about (what if the data is invalid? Also, you should at least assert the one fields. Without it, how do know, for example, if the dates were parsed correctly?)